### DISCUSSION TITLE
Make logging async

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/rosys#31] [Ideas / Feature Requests]
When we log to files the writing of the log file currently blocks the main thread execution. We should find a way to do async logging!

**Comment by rodja** [comment]
Our first idea was to use the [aiologger module](https://async-worker.github.io/aiologger/usage.html) for logging to not block the UI thread when writing to disk. Unfortunately the project seems not longer to be maintained. Links in the docs are broken and the GitHub project https://github.com/async-worker/aiologger looks quite stale. There are also some worrying issues like incompatibilities with pytest etc.
